test/tests/functional/pbs_qstat_json.py

```diff
@@ -67,22 +67,20 @@ def test_qstat_json_valid(self):
         j = Job(TEST_USER)
         j.set_sleep_time(10)
         jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': "R"}, id=jid)
         f = open('/tmp/qstat_script', 'w')
-        f.write(self.server.pbs_conf['PBS_EXEC'] +
-                "/bin/qstat -f -F json > /tmp/qstat_out")
+        f.write(os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                             'bin', 'qstat') + " -f -F json > /tmp/qstat_out")
```

Comment:
While it is highly unlikely, if two copies of this test were being run at the same time, you'd get a race condition.  One test's file would overwrite the other.  Maybe use tempfile.mkstemp()?